gateway/gateway.go

```diff
@@ -308,22 +309,27 @@ func (gateway *HandleT) dbWriterWorkerProcess(process int) {
 		for _, userWorkerBatchRequest := range breq.batchUserWorkerBatchRequest {
 			jobList = append(jobList, userWorkerBatchRequest.jobList...)
 		}
-		err := gateway.jobsDB.WithStoreSafeTx(func(tx jobsdb.StoreSafeTx) error {
-			if gwAllowPartialWriteWithErrors {
-				errorMessagesMap = gateway.jobsDB.StoreWithRetryEachInTx(tx, jobList)
-			} else {
-				err := gateway.jobsDB.StoreInTx(tx, jobList)
-				if err != nil {
-					gateway.logger.Errorf("Store into gateway db failed with error: %v", err)
-					gateway.logger.Errorf("JobList: %+v", jobList)
+
+		err := misc.RetryWith(context.Background(), gateway.jobdDBRequestTimeout, gateway.jobdDBMaxRetries, func(ctx context.Context) error {
```

Comment:
There is no point in timing out after 90s and retrying 3 times. HTTP client would have timed out by then.

Reply:
true, in gateway we need to use the same timeout as the http server uses and no retries